File: Atividades/Ordenação/C#/sorting/Quicksort.cs
using System;
using System.Runtime.CompilerServices;

namespace C_.sorting
{
    public class Quicksort
    {
        public void quicksort(List<int> roster, int start = 0, int stop = -1)
        {
            if (stop == -1)
                stop = roster.Count - 1;

            if (start < stop)
            {
                int pi = partition(roster, start, stop);
                quicksort(roster, start, pi - 1);
                quicksort(roster, pi + 1, stop);
            }
        }

        public int partition(List<int> roster, int start, int stop)
        {
            int pivot = roster[stop];
            int i = start;

            for (int j = start; j < stop; j++)
            {
                if (roster[j] >= pivot)
                {
                    (roster[j], roster[i]) = (roster[i], roster[j]);
                    i++;
                }
            }

            (roster[i], roster[stop]) = (roster[stop], roster[i]);
            return i;
        }
    }
}

File: Atividades/Ordenação/Python/Ordenações/QuickSort.py
# Método QuickSort
def quicksort(roster, start=0, stop=None):
    # Decide a parada
    if stop == None:
        stop = len(roster)-1
    # Chamada recursiva do método
    if start < stop:
        pi = partition(roster, start, stop)
        quicksort(roster, start, pi-1)
        quicksort(roster, pi+1, stop)


# Método de Partição
def partition(roster, start, stop):
    pivot = roster[stop]
    i = start
    for j in range(start, stop):
        if roster[j] >= pivot:
            roster[j], roster[i] = roster[i], roster[j]
            i += 1
    
    roster[i], roster[stop] = roster[stop], roster[i]
    return i